lib/sinks/fs.js

```diff
@@ -36,18 +36,44 @@ class SinkFS {
     }
 
     read(filePath) {
-        const pathname = path.join(this._rootPath, filePath);
-        return fs.createReadStream(pathname, {
-            autoClose: true,
-            emitClose: true,
+        return new Promise((resolve, reject) => {
+            const pathname = path.join(this._rootPath, filePath);
+
+            if (pathname.indexOf(this._rootPath) !== 0) {
+                return reject(new Error(`Directory traversal - ${filePath}`));
```

Comment:
More descriptive error here? Different error type maybe?

Reply:
This is an extra check to prevent that someone tries to walk the directory through providing stuff like `../../../file/system/secret.file` in the URLs (a check for `../` should be done on all parameters in the HTTP routes also to prevent the same way further up).

I am not quite sure what more to write of what type of Error to use.

This should be logged though.